gcloud-java-compute/src/main/java/com/google/gcloud/compute/Compute.java

```diff
@@ -24,4 +30,548 @@
  * @see <a href="https://cloud.google.com/compute/docs/">Google Cloud Compute Engine</a>
  */
 public interface Compute extends Service<ComputeOptions> {
+
+  /**
+   * Fields of a Compute Engine DiskType resource.
+   *
+   * @see <a href="https://cloud.google.com/compute/docs/reference/latest/diskTypes#resource">Disk
+   *     Type Resource</a>
+   */
+  enum DiskTypeField {
+    CREATION_TIMESTAMP("creationTimestamp"),
+    DEFAULT_DISK_SIZE_GB("defaultDiskSizeGb"),
+    DESCRIPTION("description"),
+    ID("id"),
+    NAME("name"),
+    SELF_LINK("selfLink"),
+    VALID_DISK_SIZE("validDiskSize"),
+    ZONE("zone");
+
+    private final String selector;
+
+    DiskTypeField(String selector) {
+      this.selector = selector;
+    }
+
+    public String selector() {
```

Comment:
I am not sure if we discussed that already.
If we don't mind to make this method `public` than we could make all the XXXField enums implement a "Selector" interface and have one static `String selector(Selector required, Selector... other)` method.